### ISSUE TITLE
Use pundit's `policy_scope` instead of administrate's `policy_scope_admin`

### ISSUE THREAD

**Issue by robinboening** [thoughtbot/administrate#1411]
Hey all,

thank you for providing and maintaining this nice piece of software! I really appreciate your efforts. 

What is this issue about:

I am using Pundit's `policy_scope` in controller's index actions to make sure users can only access their own records. To make sure no one ever forgets using it I use this neat little trick provided by Pundit: `after_action :verify_policy_scoped, only: :index` - It raises an error if I haven't used `policy_scope` in my `index` actions.

Administrate is not using `policy_scope`. Instead it circumvents using it by calling a method `policy_scope_admin` provided by administrate which basically does the same but introduces the `resolve_admin` method.

This prevents me from using `after_action :verify_policy_scoped, only: :index` in the admin namespace and made me question if `resolve_admin` is really worth circumventing Pundits `policy_scope` method.

In my eyes there is no benefit of using `resolve_admin` over only using the default `resolve` method compared in these examples:

```
class Scope < Scope
  def resolve_admin
    scope.all
  end

  def resolve
    scope.where(user_id: user.id)
  end
end
```

```
class Scope < Scope
  def resolve
    if user.admin?
      scope.all
    else
      scope.where(user_id: user.id)
    end
  end
end
```

I am questioning is the `resolve_admin` and with that the `policy_scope_admin` method really needed? In my eyes it provides very little benefit but prevents me from using Pundit's full toolset.

What do you think?

**Comment by robinboening** [comment]
I had a chance to think about it again. I understand my two examples might look similar, but differ in an important aspect: The first simply depends on the user role while using `resolve_admin` makes sure we are in administrate context (independent of user roles).

I now see the use case for `resolve_admin`, but I still don't like the fact administrate is not using the `policy_scope` method.

What do you think about this:

1.) Pundit should provide a way of registering `resolve-*` methods for the scope.
2.) administrate uses that interface instead of implementing its own method.

If you agree with this I could walk to pundit and see if they like the idea as well.

**Comment by robinboening** [comment]
Today I was digging deeper into pundit and found a way to solve all my issues at once. I also believe this is a more elegant way as currently implemented in administrate.

What did I solve:

✅I don't want to use `resolve_admin` in my policies
✅I want to use Pundit's `policy_scope` method so I can verify the use of it with `verify_policy_scoped`
✅I want my policies to be very simple with the least logic possible

I namespaced my policies. Every policy relevant for administrate now lives in `policies/administrate/`. Every policy relevant for the frontend remains or also lives in `policies/`. Yes, I might have two policies for one model, but separated by concerns which makes them simpler because they only need to reflect one concern.

To make this work I had to overwrite three methods coming from `punditize.rb`

```
def scoped_resource
  policy_scope([:administrate, resource_class.default_scoped])
end

def show_action?(action, resource)
  Pundit.policy!(pundit_user, [:administrate, resource]).send("#{action}?".to_sym)
end

def authorize(record, query = nil)
  super([:administrate, record], query)
end
```

Thats it. I love this pattern and like to suggest changing `punditize.rb` this way. I could send a PR if you agree?
